### DISCUSSION TITLE
Use rapier with an animated glb ?

### DISCUSSION THREAD

**Discussion by ThomasCartier** [dimforge/rapier#556] [Q&A] [answered]
Hi,
I intend to create a simulation of falling cubes on an animated character.

Reading [the getting started](https://rapier.rs/docs/user_guides/rust/getting_started#basic-simulation-example), and [colliders](https://rapier.rs/docs/user_guides/rust/colliders) I understand that I should update the character animation position at each tick myself (calculate final vertices position and resubmit a buffer of positions) and submit the updated Trimesh buffer to the proper collider in  [the collider set](https://docs.rs/rapier3d/latest/rapier3d/geometry/struct.ColliderSet.html). And keep going up to the end of the anim.

Is that it ?
Sorry if my question is obvious :+1:

**Comment by doppl3r** [comment]
From the collider's documentation: "It is discouraged to use a triangle meshes or a polylines for colliders attached to dynamic rigid-bodies." I'd also assume recreating a trimesh buffer every frame would be incredibly expensive over time.

Have you tried creating your own ragdoll using joints? If your character has arms that need to interact with other rigid bodies, then you can use cuboids or capsules colliders for parts of the arm (upper + forearm). If your model is using bones, you can copy the position + rotation of each bone for all colliders.

**Reply by ThomasCartier** [reply]
Ok, I will go with a compound shape and use the joint accordingly. 

The humanoid won't be affected by the cubes falling on it. Instead, the cubes will be. That animated humanoid just acts as an obstacle for the falling cubes and shows how cubes could react with different poses.

The performance is secondary (but stil lmatters) because it is calculated offline.

So I should only update the rotations and positions of each bone in their respective colliders, right ? (That's a LOT less work for me as well, since I already have the animation data through the glb).

Am I correct ? (just to make sure I got it)
thanks !

**Reply by doppl3r** [reply, marked as answer]
That is correct, the documentation will tell you how to update the collider position/rotation after it is created:
https://rapier.rs/docs/user_guides/javascript/colliders

**Reply by ThomasCartier** [reply]
Thanks a lot, you were very helpful !
